### PR TITLE
Performance analysis vignette

### PR DESCRIPTION
**Pull request by pssachdeva**
Contains Rmarkdown file for performance analysis vignette. This includes the addition of another dataset - East Ramapo 2018 Voter file - which as been modified as such:

- Voter ID has been replaced
- Surnames have been exchanged with their "closest" matching surname in terms of racial probabilities
- Pre-spatial joined into Census blocks and wards in the district map.

### CODE REVIEW

File: vignettes/performance_analysis.Rmd [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+  library(eiCompare)
+  library(ggplot2)
+  library(sf)
+  library(wru)
+})
+```
+
+## What is a performance analysis? 
+
+A successful voting rights case may result in a local jurisdiction's districting map to be thrown out, prompting the need for a new map. Thus, it's necessary assess whether the new map provides sufficient representation for minority groups. It’s not desirable to simply wait for elections to happen and see what the results might be. Instead, we can look at past elections, and observe how candidates would perform if the proposed map was used at the time of the election. This is the basis of a performance analysis.
```
Comment (aridf) [inline vignettes/performance_analysis.Rmd:23]: A successful voting rights case may result in a local jurisdiction's districting map being thrown out

File: vignettes/performance_analysis.Rmd [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+4. **Aggregation:** Aggregate the race estimates per district to obtain voter turnout, by race, in each district.
+
+`eiCompare` provides functions to perform steps 2-4 of the analysis, as well as a function that completes the entire pipeline. Note that Gecoding (step 1) must be performed separately (`eiCompare` provides tools to aid in geocoding: see the Geocoding vignette). In the following section, we walk through each of the steps.
+
+## Case study: East Ramapo School District
+
+The example we'll use to demonstrate the performance analysis is East Ramapo School District (ERSD), located in Rockland County of the New York City suburbs. ERSD is highly segregated, with the majority of Black and Hispanic students attending public schools and white students attending private schools. Furthermore, its voting system for the School Board was at-large, where all voters could vote for all seats on the school board. This system favored the white families whose students largely attended private schools, resulting in redistribution of funds that had adverse impacts on public school students.
```
Comment (aridf) [inline vignettes/performance_analysis.Rmd:40]: Furthermore, ERSD uses an at-large voting system for School Board elections, where all voters could...

File: vignettes/performance_analysis.Rmd [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+  guides(fill = guide_legend(
+    title = "Fraction\nMinority",
+    title.position = "top",
+    title.size = 10
+  ))
+show(cvap_map)
+```
+
+Examining this map reveals that, according to CVAP, four wards (1-4) have potential for minority voters to elect a representative of their choice (they have a plurality or majority). However, due to turnout differences across racial groups, this does not imply that the minority voters would have actually turned out sufficiently enough to elect four representatives. In other words, this map may not have "performed" well enough to guarantee representation, due to turnout.
+
```
Comment (aridf) [inline vignettes/performance_analysis.Rmd:81]: Might want to add a transition sentence here like "Next, we'll walk through how to conduct a performance analysis and test this hypothesis".

File: vignettes/performance_analysis.Rmd [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+Ultimately, we need to assess the demographic breakdown across racial groups for each district in the new map. What data source do we use to calculate the percentage of each racial group? We could use total population or citizen voting age population (CVAP), as provided by the Census Bureau. However, these measures may not be accurate, because they assume that voter turnout is equal across racial groups. This is not always true, especially in cases of racial gerrymandering, where turnout for minority groups may be depressed. Instead, we argue that going to the level of the _voter file_ is necessary, as the voter file actually informs _who turned out to vote_.
+
+## The steps in a performance analysis
+
+To conduct a performance analysis, we need to join the voters in a voter file to the new districts and determine the turnout by race, per district. This includes the following steps:
+
+1. **Geocoding:** Geocode latitude and longitude of voters in the file using their address. We do not focus on this step in this vignette.
```
Comment (aridf) [inline vignettes/performance_analysis.Rmd:31]: In the website version (if we're permitting ourselves small differences), could link to Juandalyn's vignette here.

File: vignettes/performance_analysis.Rmd [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+```{r race_probs}
+print(bisg$bisg[, c(
+  "voterid",
+  "surname",
+  "pred.whi",
+  "pred.bla",
+  "pred.his",
+  "pred.asi"
+)])
+```
```
Comment (aridf) [inline vignettes/performance_analysis.Rmd:201]: Maybe worth really slowly walking through how to interpret these probabilities, eg. "0.71 means the person has a 71% probability of being race X" but with better wording. Just so you have some description of this output before moving on to the next thing.

**Comment by pssachdeva** [comment]
@scottyhq I'm a bit bewildered, because adding the line
```
data(ersd_maps)
st_transform(ersd_maps) <- 4326
```
fixed the issue. What's confusing is that a previous commit had already changed the dataset `ersd_maps` to this CRS, but didn't solve this problem. I don't think adding this line should really be changing the CRS at all, but it helped solved the CMD check pass...

File: vignettes/performance_analysis.Rmd [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+
+### Spatial join between the voter and shape files
+
+Next, we need to identify which wards these voters are located in. Performing this spatial join is abstracted in the `merge_voter_file_to_shape()` function, which can convert the voter file to a geometry object and join on location. This function, in addition to requiring the voter and shape files, also requires a Coordinates Reference System to use (as a PROJ4 string) as well as the column names for the longitude, latitude, and voter ID (in order to de-duplicate the voter file after the spatial join).
+
+```{r merge_to_ward}
+voter_file_w_ward <- eiCompare::merge_voter_file_to_shape(
+  voter_file = voter_file,
+  shape_file = ersd_maps,
+  crs = "+proj=longlat +ellps=GRS80",
```
[the remark this answers is not in the document]
Comment (pssachdeva) [reply vignettes/performance_analysis.Rmd]: I ended up allowing either an input CRS, or using the shape file's CRS (default option).

**Pull request merged**